src/relay/relay.py

```diff
@@ -175,10 +177,22 @@ def get_event_selector_for_currency_network(self, network_address=None):
                 from_to_types=currency_network_events.from_to_types,
             )
         else:
-            if network_address is None:
-                return Proxy(self._web3, abi=self.contracts["CurrencyNetwork"]["abi"])
             return self.currency_network_proxies[network_address]
 
+    def get_event_selector_for_proxy(self) -> Union[ethindex_db.EthindexDB, Proxy]:
+        """return either a Proxy or a EthindexDB instance
+        This is being used from relay.api to query for events.
+        """
+        if self.use_eth_index:
+            return ethindex_db.EthindexDB(
+                ethindex_db.connect(""),
+                standard_event_types=currency_network_events.standard_event_types,
+                event_builders=currency_network_events.event_builders,
+                from_to_types=currency_network_events.from_to_types,
+            )
+        else:
+            return Proxy(self._web3, abi=self.contracts["CurrencyNetwork"]["abi"])
```

Comment:
I don't think it is clear that you ask for a general proxy and get a proxy which only knows the abi of currency networks.

Reply:
Idk, I tried to clean up some of the mess that we sometimes dont return a CurrencyNetworkProxy in the `get_event_selector_for_currency_network`
I'll try to give it a better name.

Reply:
I like the new name better.